From 9cfe0df35ab1b486735252c54f2ea98906366831 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Thu, 21 Sep 2023 09:41:29 +0200
Subject: [PATCH 0513/1695] dts: stm32: add nvram (NVMEM) node under tamp in
 stm32mp251.dtsi

Backup Registers nvram will be accessed using the NVMEM framework

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9083699fc9d9153aebfdf4a3f626906468f030bd dts: stm32: add nvram (NVMEM) node under tamp in stm32mp251.dtsi]
Change-Id: If0efce97f5ecd845844aa1b23233971be200ae65
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/328930
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
(cherry picked from commit dec376aab958badf4781ba5b8a458abdfce258e8)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351778
---
 core/arch/arm/dts/stm32mp251.dtsi | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 395de172a..d55119da2 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -763,7 +763,18 @@
 			interrupts = <GIC_SPI 14 IRQ_TYPE_LEVEL_HIGH>;
 			#address-cells = <1>;
 			#size-cells = <1>;
-			status = "disabled";
+			ranges;
+
+			nvram: nvram@46010100 {
+				compatible = "st,stm32mp25-tamp-nvram";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <0x46010100 0x200>;
+
+				boot_mode: tamp-bkp@180 {
+					reg = <0x180 0x4>;
+				};
+			};
 		};
 
 		pinctrl_z: pinctrl-z@46200000 {
-- 
2.47.2

